pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -93,7 +93,7 @@
     protected final ClientConfigurationData conf;
     private LookupService lookup;
     private final ConnectionPool cnxPool;
-    private final Timer timer;
+    private static final Timer timer = new HashedWheelTimer(getThreadFactory("pulsar-timer"), 1, TimeUnit.MILLISECONDS);;
```

Comment:
if you make this field `static` then you cannot stop it when you stop the PulsarClient.